### ISSUE TITLE
find function not working in 3.1.0

### ISSUE THREAD

**Issue by dobrysmak** [lodash/lodash#982]
Hi,
we noticed that find function with given object is returning *undefined* in any case
```
_.find(arrayOfObjects, { key: 'mykeyvalue' });
```
yet it works fine when we pass function as search param
```
_.find(arrayOfObjects, function (obj) { return obj.key === 'mykeyvalue' });
```

**Comment by jdalton** [comment]
I suspect you're seeing the result of the v3.0 [change](https://github.com/lodash/lodash/wiki/Changelog#v300) to iterate over own properties of objects when comparing. You can use the alternative `_.matchesProperty` shorthand instead:

```js
_.find(arrayOfObjects, 'key', 'mykeyvalue');
```

**Comment by jdalton** [comment]
The shorthand is available in v3.2.0.
